frontend-v2/src/context/ChatContext.tsx

```diff
@@ -0,0 +1,35 @@
+import { FC, PropsWithChildren, createContext, useState } from 'react';
+import { useSocket } from '../hooks/socket';
+
+export type IChat = {
+  id: string;
+  name: string;
+  type: 'public' | 'private' | 'protected by password';
+  createdAt: string;
+  updatedAt: string;
+  lastMessage?: string;
+  avatar?: string;
+};
+
+export type IChatContext = {
+  activeChat: IChat | null;
+  setActiveChat: (chat: IChat | null) => void;
+};
+
+export const ChatContext = createContext<IChatContext | null>(null);
+
+export const ChatContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [messages, setMessages] = useState<string[]>([]);
+  const [activeChat, setChat] = useState<IChat | null>(null);
```

Comment:
would this be `[activeChat, setActiveChat]` ? just to maintain the pattern of `useState`

Reply:
no need for this :)